### PR TITLE
Make commit classifier apply the phabricator patch

### PR DESCRIPTION
**Pull request by bhushan-borole**
Commit classifier now applies the phabricator patch as its first step(right after the cloning).
This PR resolves issue #1268

### CODE REVIEW

**Comment by marco-c** [comment]
> I am facing a problem for the TaskGroup checks, "bugbug link" task has failed while rest of them has passed.
Can you tell me how to resolve this problem?

> Logs : https://community-tc.services.mozilla.com/tasks/MsFDby5-QRSxkFM04ed4Tg/runs/0/logs/https%3A%2F%2Fcommunity-tc.services.mozilla.com%2Fapi%2Fqueue%2Fv1%2Ftask%2FMsFDby5-QRSxkFM04ed4Tg%2Fruns%2F0%2Fartifacts%2Fpublic%2Flogs%2Flive.log

It's `bugbug lint`, it is the linting task.
You can use pre-commit to handle linting, it is described in the README: https://github.com/mozilla/bugbug#auto-formatting.

File: scripts/commit_classifier.py [hunk 1]
```diff
@@ -552,31 +577,12 @@ def feature_sort_key(f):
...
-        with hglib.open(self.repo_dir) as hg:
-            if phabricator_deployment is not None and diff_id is not None:
-                self.apply_phab(hg, phabricator_deployment, diff_id)
-
-                revision = hg.log(revrange="not public()")[0].node.decode("utf-8")
-
-            # Analyze patch.
-            commits = repository.download_commits(
-                self.repo_dir, rev_start=revision, save=False
-            )
```
Comment (marco-c) [inline scripts/commit_classifier.py:578]: This won't work, as the `commits` variable is used later in the function.

Comment (bhushan-borole) [reply scripts/commit_classifier.py:578]: I got my mistake, resolved it.

File: scripts/commit_classifier.py [hunk 2]
```diff
@@ -146,6 +155,23 @@ def __init__(self, model_name, repo_dir, git_repo_dir, method_defect_predictor_d
...
+
+        if diff_id is not None:
+            assert phabricator_deployment is not None
+            assert revision is None
+
+        self.update_commit_db()
+
+        with hglib.open(self.repo_dir) as hg:
+            if phabricator_deployment is not None and diff_id is not None:
+                self.apply_phab(hg, phabricator_deployment, diff_id)
```
Comment (marco-c) [inline scripts/commit_classifier.py:171]: We could move the `download_and_load_model` call after this, so we don't download the models needlessly if we can't apply the patch.

File: scripts/commit_classifier.py [hunk 3]
```diff
@@ -146,6 +155,23 @@ def __init__(self, model_name, repo_dir, git_repo_dir, method_defect_predictor_d
...
+
+                self.revision = hg.log(revrange="not public()")[0].node.decode("utf-8")
```
Comment (marco-c) [inline scripts/commit_classifier.py:173]: You can move all this code in the `if diff_id is not None:` branch above

**Comment by marco-c** [comment]
You can test the script by running a command such as:
```
BUGBUG_PHABRICATOR_URL='https://phabricator.services.mozilla.com/api/' BUGBUG_PHABRICATOR_TOKEN="YOUR_PHABRICATOR_API_KEY" PYTHONPATH=. python3 -c "from scripts import commit_classifier; commit_classifier.main()" regressor PATH_TO_MOZILLA_CENTRAL/mozilla-central --diff-id=234724 --phabricator-deployment=prod
```
and:
```
BUGBUG_TEST_FAILURE_CONFIDENCE_THRESHOLD="0.7" BUGBUG_TEST_SELECTION_CONFIDENCE_THRESHOLD="0.3" BUGBUG_PHABRICATOR_URL='https://phabricator.services.mozilla.com/api/' BUGBUG_PHABRICATOR_TOKEN="YOUR_PHABRICATOR_API_KEY" PYTHONPATH=. python3 -c "from scripts import commit_classifier; commit_classifier.main()" testlabelselect PATH_TO_MOZILLA_CENTRAL/mozilla-central --revision='444e0f025cfa' --runnable-jobs='https://firefox-ci-tc.services.mozilla.com/api/queue/v1/task/YZOfqacTTIyUapQK3XrluQ/runs/0/artifacts/public/runnable-jobs.json'
```

File: scripts/commit_classifier.py (hunk 2 above)
Comment (bhushan-borole) [reply scripts/commit_classifier.py:171]: This means, if phabricator patch is not applied, then model will not be downloaded.
Sure will work on this step.

File: scripts/commit_classifier.py (hunk 3 above)
Comment (bhushan-borole) [reply scripts/commit_classifier.py:173]: `if diff_id is not None:` above`with hglib.open` ?

Comment (bhushan-borole) [reply scripts/commit_classifier.py:173]: But we will still check for the `phabricator_deployment` variable right ?

**Comment by bhushan-borole** [comment]
I am getting error in the pre-commit black hook, can you tell me please what it is?

File: scripts/commit_classifier.py [hunk 4]
```diff
@@ -146,6 +152,29 @@ def __init__(self, model_name, repo_dir, git_repo_dir, method_defect_predictor_d
...
+            with hglib.open(self.repo_dir) as hg:
+                if phabricator_deployment is not None:
+                    self.apply_phab(hg, phabricator_deployment, diff_id)
+
+                    self.revision = hg.log(revrange="not public()")[0].node.decode(
+                        "utf-8"
+                    )
+
+                    # model will get downloaded only if patch is applied
+                    self.model = download_and_load_model(model_name)
```
Comment (marco-c) [inline scripts/commit_classifier.py:175]: The model needs to be downloaded also if revision is not None and diff_id is None, so it should be outside the `         if diff_id is not None:`

Comment (marco-c) [reply scripts/commit_classifier.py:175]: If the patch fails to apply, we'll stop here anyway, so we won't reach the download code.

File: scripts/commit_classifier.py [hunk 5]
```diff
@@ -550,33 +579,13 @@ def feature_sort_key(f):
...
-
-                revision = hg.log(revrange="not public()")[0].node.decode("utf-8")
-
-            # Analyze patch.
-            commits = repository.download_commits(
-                self.repo_dir, rev_start=revision, save=False
-            )
-
+        # Analyze patch.
+        rev_start = self.revision
```
Comment (marco-c) [inline scripts/commit_classifier.py:585]: Nit: no need to store self.revision in another variable, just use `self.revision` where needed

**Comment by marco-c** [comment]
> You can test the script by running a command such as:

Did you try to test the script?

File: scripts/commit_classifier.py (hunk 4 above)
Comment (bhushan-borole) [reply scripts/commit_classifier.py:175]: But here, we don't want to download the model, if we don't apply the patch right?
https://github.com/mozilla/bugbug/pull/1697#discussion_r455419168

**Comment by bhushan-borole** [comment]
> > You can test the script by running a command such as:
> 
> Did you try to test the script?

I am trying to login at https://phabricator.services.mozilla.com
But I am getting an exception on the page.

File: scripts/commit_classifier.py (hunk 4 above)
Comment (bhushan-borole) [reply scripts/commit_classifier.py:175]: Oh okay, got it.
Will fix it right away, thank you

File: scripts/commit_classifier.py (hunk 5 above)
Comment (bhushan-borole) [reply scripts/commit_classifier.py:585]: Fixed

**Comment by bhushan-borole** [comment]
Hello @marco-c.
Can you tell me how to access the website for API Key so that I can run that command?
Thank you.

**Comment by bhushan-borole** [comment]
Tagging you @marco-c 
Incase you missed

**Comment by marco-c** [comment]
>  Can you tell me how to access the website for API Key so that I can run that command?

Here are the steps required: https://moz-conduit.readthedocs.io/en/latest/phabricator-user.html#creating-an-account.

**Comment by bhushan-borole** [comment]
Hello @marco-c 
I am getting a 404 page, when running the command `moz-phab install-certificate` while generating the API key.
Attaching Screenshots for reference:
Capture
Capture

**Comment by marco-c** [comment]
Don't worry about this error, you now have an account and you can get your API key from https://phabricator.services.mozilla.com/settings/user/YOURUSERNAME/page/apitokens/ (you can reach this page by clicking on your user image on the top right of https://phabricator.services.mozilla.com/, then in the menu that appears select "Settings", then in the left side select "Conduit API Tokens").

**Comment by bhushan-borole** [comment]
I ran the first command by adding the API token, I am facing this error:

```
Traceback (most recent call last):
  File "<string>", line 1, in <module>
  File "/home/borolebhushan8/bugbug/scripts/commit_classifier.py", line 793, in main
    args.diff_id,
  File "/home/borolebhushan8/bugbug/scripts/commit_classifier.py", line 160, in __init__
    self.update_commit_db()
  File "/home/borolebhushan8/bugbug/scripts/commit_classifier.py", line 246, in update_commit_db
    repository.clone(self.repo_dir, update=True)
  File "/home/borolebhushan8/bugbug/bugbug/repository.py", line 1113, in clone
    with hglib.open(repo_dir) as hg:
  File "/home/borolebhushan8/.local/lib/python3.6/site-packages/hglib/__init__.py", line 11, in open
    return client.hgclient(path, encoding, configs)
  File "/home/borolebhushan8/.local/lib/python3.6/site-packages/hglib/client.py", line 75, in __init__
    self.open()
  File "/home/borolebhushan8/.local/lib/python3.6/site-packages/hglib/client.py", line 272, in open
    self.server = util.popen(self._args, self._env)
  File "/home/borolebhushan8/.local/lib/python3.6/site-packages/hglib/util.py", line 241, in popen
    startupinfo=startupinfo, env=environ)
  File "/usr/lib/python3.6/subprocess.py", line 729, in __init__
    restore_signals, start_new_session)
  File "/usr/lib/python3.6/subprocess.py", line 1364, in _execute_child
    raise child_exception_type(errno_num, err_msg, err_filename)
FileNotFoundError: [Errno 2] No such file or directory: 'hg': 'hg'
```

**Comment by marco-c** [comment]
It looks like you don't have Mercurial installed on your system. Did you install all requirements in requirements.txt?
`python -m pip install -r requirements.txt`

What is the output of `python -m pip show mercurial`?

**Comment by bhushan-borole** [comment]
I resolved the previous error, and I am getting a new one:

```
Traceback (most recent call last):
  File "<string>", line 1, in <module>
  File "/home/borolebhushan8/bugbug/scripts/commit_classifier.py", line 793, in main
    args.diff_id,
  File "/home/borolebhushan8/bugbug/scripts/commit_classifier.py", line 160, in __init__
    self.update_commit_db()
  File "/home/borolebhushan8/bugbug/scripts/commit_classifier.py", line 246, in update_commit_db
    repository.clone(self.repo_dir, update=True)
  File "/home/borolebhushan8/bugbug/bugbug/repository.py", line 1143, in clone
    subprocess.run(cmd, check=True)
  File "/usr/lib/python3.6/subprocess.py", line 438, in run
    output=stdout, stderr=stderr)
subprocess.CalledProcessError: Command '['hg', 'robustcheckout', b'--purge', b'--sharebase=PATH_TO_MOZILLA_CENTRAL/mozilla-central-shared', b'--networkattempts=7', b'--branch=tip', b'--', 'https://hg.mozilla.org/mozilla-central', 'PATH_TO_MOZILLA_CENTRAL/mozilla-central']' returned non-zero exit status 255.
```

**Comment by marco-c** [comment]
You'll need to run steps 2 and 3 from https://github.com/mozilla/bugbug#running-the-repository-mining-script.

**Comment by bhushan-borole** [comment]
How do I install the `mozci` package, it says:
```
ERROR: Could not find a version that satisfies the requirement mozci (from versions: none)
ERROR: No matching distribution found for mozci
```

**Comment by marco-c** [comment]
@bhushan-borole I've never seen that error before, what version of Python are you using?

**Comment by bhushan-borole** [comment]
3.6+

**Comment by bhushan-borole** [comment]
> 3.6+

I updated the version, that installed all the requirements correctly.
Another error came up:

```
Traceback (most recent call last):
  File "<string>", line 1, in <module>
ModuleNotFoundError: No module named 'scripts'
```

**Comment by bhushan-borole** [comment]
> 3.6+

Hello @marco-c 
I updated the python version to 3.7, and all the requirements are installed successfully now.
But I am facing a new error:

```
*** failed to import extension absorb: No module named absorb
*** failed to import extension evolve from /home/borolebhushan8/.mozbuild/evolve/hgext3rd/evolve: No module named utils.stringutil
*** failed to import extension firefoxtree from /home/borolebhushan8/.mozbuild/version-control-tools/hgext/firefoxtree: No module named wireprotov1server
(third party extension push-to-try requires version 4.6 or newer of Mercurial; disabling)
hg: unknown command 'robustcheckout'
(did you mean checkout?)
Traceback (most recent call last):
  File "<string>", line 1, in <module>
  File "/home/borolebhushan8/bugbug/scripts/commit_classifier.py", line 793, in main
    args.diff_id,
  File "/home/borolebhushan8/bugbug/scripts/commit_classifier.py", line 160, in __init__
    self.update_commit_db()
  File "/home/borolebhushan8/bugbug/scripts/commit_classifier.py", line 246, in update_commit_db
    repository.clone(self.repo_dir, update=True)
  File "/home/borolebhushan8/bugbug/bugbug/repository.py", line 1143, in clone
    subprocess.run(cmd, check=True)
  File "/usr/lib/python3.7/subprocess.py", line 512, in run
    output=stdout, stderr=stderr)
subprocess.CalledProcessError: Command '['hg', 'robustcheckout', b'--purge', b'--sharebase=/home/borolebhushan8/mozilla-central-shared', b'--networkattempts=7', b'--branch=tip', b'--', 'https://hg.mozilla.org/mozilla-central', '/home/borolebhushan8/mozilla-central']' returned non-zero exit status 255.
```

**Comment by bhushan-borole** [comment]
> > But I am getting an error here also:
> 
> You need to run Step 2 and Step 3 (make sure you enable all the extensions mentioned in infra/hgrc!) only, not Step 4.
> 
> Once Step 3 is done, the other error you mentioned should be gone.

Which revision should I specify here:
```
usage: repository.py [-h] repository_dir rev_start
repository.py: error: the following arguments are required: rev_start
```

**Comment by marco-c** [comment]
@bhushan-borole again, you only need to run Step 2 and Step 3, not Step 4 :)

**Comment by bhushan-borole** [comment]
Getting this new error while running that command:
```
Traceback (most recent call last):
  File "<string>", line 1, in <module>
  File "/home/borolebhushan8/.local/lib/python3.7/site-packages/scripts/commit_classifier.py", line 781, in main
    args.model, args.repo_dir, args.git_repo_dir, args.method_defect_predictor_dir
  File "/home/borolebhushan8/.local/lib/python3.7/site-packages/scripts/commit_classifier.py", line 132, in __init__
    self.model = download_and_load_model(model_name)
  File "/home/borolebhushan8/.local/lib/python3.7/site-packages/bugbug/utils.py", line 234, in download_and_load_model
    path = download_model(model_name)
  File "/home/borolebhushan8/.local/lib/python3.7/site-packages/bugbug/utils.py", line 225, in download_model
    updated = download_check_etag(url)
  File "/home/borolebhushan8/.local/lib/python3.7/site-packages/bugbug/utils.py", line 193, in download_check_etag
    r.raise_for_status()
  File "/home/borolebhushan8/.local/lib/python3.7/site-packages/requests/models.py", line 941, in raise_for_status
    raise HTTPError(http_error_msg, response=self)
requests.exceptions.HTTPError: 404 Client Error: Not Found for url: https://community-tc.services.mozilla.com/api/index/v1/task/project.relman.bugbug.train_regressor.v0.0.333/artifacts/public/regressormodel.zst
```

**Comment by marco-c** [comment]
@bhushan-borole you can avoid that by setting the `TAG` environment variable to `latest`:
```
export TAG=latest
```
then running the command

**Comment by bhushan-borole** [comment]
> @bhushan-borole you can avoid that by setting the `TAG` environment variable to `latest`:
> 
> ```
> export TAG=latest
> ```
> 
> then running the command

Tried this, but getting a new error, but it is similar
```
2020-08-03 14:50:27,837:INFO:bugbug.db:Downloading https://community-tc.services.mozilla.com/api/index/v1/task/project.relman.bugbug.data_commits.latest/artifacts/public/commit_experiences.lmdb.tar.zst to data/commit_experiences.lmdb.tar.zst
tar (child): zstd: Cannot exec: No such file or directory
tar (child): Error is not recoverable: exiting now
tar: Child returned status 2
tar: Error is not recoverable: exiting now
Traceback (most recent call last):
  File "<string>", line 1, in <module>
  File "/home/borolebhushan8/bugbug/scripts/commit_classifier.py", line 786, in main
    classifier = CommitClassifier(
  File "/home/borolebhushan8/bugbug/scripts/commit_classifier.py", line 160, in __init__
    self.update_commit_db()
  File "/home/borolebhushan8/bugbug/scripts/commit_classifier.py", line 248, in update_commit_db
    assert db.download(repository.COMMITS_DB, support_files_too=True)
  File "/home/borolebhushan8/bugbug/bugbug/db.py", line 101, in download
    successful |= download_support_file(path, support_file, extract)
  File "/home/borolebhushan8/bugbug/bugbug/db.py", line 70, in download_support_file
    utils.extract_file(path)
  File "/home/borolebhushan8/bugbug/bugbug/utils.py", line 291, in extract_file
    extract_tar_zst(path)
  File "/home/borolebhushan8/bugbug/bugbug/utils.py", line 284, in extract_tar_zst
    subprocess.run(["tar", "-I", "zstd", "-xf", path], check=True)
  File "/usr/lib/python3.8/subprocess.py", line 512, in run
    raise CalledProcessError(retcode, process.args,
subprocess.CalledProcessError: Command '['tar', '-I', 'zstd', '-xf', 'data/commit_experiences.lmdb.tar.zst']' returned non-zero exit status 2.
```

**Comment by marco-c** [comment]
You need to install zstandard on your machine (`apt install zstd`, if you're on a Debian-based distribution).

**Comment by bhushan-borole** [comment]
Do you also need `rust-code-analysis` ?

**Comment by marco-c** [comment]
Yes! You'll need to install Rust, and then run `cargo install rust-code-analysis-cli`.

**Comment by marco-c** [comment]
Or do the same we're doing in the Dockerfile: https://github.com/mozilla/bugbug/blob/ed52b6de51e2f6e34c2c0103b0251bf416d53fa0/infra/dockerfile.commit_retrieval#L24.

**Comment by bhushan-borole** [comment]
I installed, but still I am getting the same error
```
Traceback (most recent call last):
  File "/home/borolebhushan8/bugbug/bugbug/rust_code_analysis_server.py", line 47, in start_process
    self.proc = subprocess.Popen(
  File "/usr/lib/python3.8/subprocess.py", line 854, in __init__
    self._execute_child(args, executable, preexec_fn, close_fds,
  File "/usr/lib/python3.8/subprocess.py", line 1702, in _execute_child
    raise child_exception_type(errno_num, err_msg, err_filename)
FileNotFoundError: [Errno 2] No such file or directory: 'rust-code-analysis'

During handling of the above exception, another exception occurred:

Traceback (most recent call last):
  File "<string>", line 1, in <module>
  File "/home/borolebhushan8/bugbug/scripts/commit_classifier.py", line 795, in main
    classifier.classify(args.runnable_jobs)
  File "/home/borolebhushan8/bugbug/scripts/commit_classifier.py", line 583, in classify
    commits = repository.download_commits(
  File "/home/borolebhushan8/bugbug/bugbug/repository.py", line 1047, in download_commits
    code_analysis_server = rust_code_analysis_server.RustCodeAnalysisServer()
  File "/home/borolebhushan8/bugbug/bugbug/rust_code_analysis_server.py", line 24, in __init__
    self.start_process()
  File "/home/borolebhushan8/bugbug/bugbug/rust_code_analysis_server.py", line 51, in start_process
    raise Exception("rust-code-analysis is required for code analysis")
Exception: rust-code-analysis is required for code analysis
```

**Comment by marco-c** [comment]
It looks like the binary name was changed. You'll need to change https://github.com/mozilla/bugbug/blob/ed52b6de51e2f6e34c2c0103b0251bf416d53fa0/bugbug/rust_code_analysis_server.py#L48 and replace `rust-code-analysis` with `rust-code-analysis-cli`.

**Comment by bhushan-borole** [comment]
```
Traceback (most recent call last):
  File "<string>", line 1, in <module>
  File "/home/borolebhushan8/bugbug/scripts/commit_classifier.py", line 795, in main
    classifier.classify(args.runnable_jobs)
  File "/home/borolebhushan8/bugbug/scripts/commit_classifier.py", line 587, in classify
    self.classify_regressor(commits)
  File "/home/borolebhushan8/bugbug/scripts/commit_classifier.py", line 596, in classify_regressor
    commits[-1],
IndexError: tuple index out of range
```

**Comment by bhushan-borole** [comment]
> It looks like the binary name was changed. You'll need to change
> 
> https://github.com/mozilla/bugbug/blob/ed52b6de51e2f6e34c2c0103b0251bf416d53fa0/bugbug/rust_code_analysis_server.py#L48
> 
> and replace `rust-code-analysis` with `rust-code-analysis-cli`.

This helped!

**Comment by bhushan-borole** [comment]
> ```
> Traceback (most recent call last):
>   File "<string>", line 1, in <module>
>   File "/home/borolebhushan8/bugbug/scripts/commit_classifier.py", line 795, in main
>     classifier.classify(args.runnable_jobs)
>   File "/home/borolebhushan8/bugbug/scripts/commit_classifier.py", line 587, in classify
>     self.classify_regressor(commits)
>   File "/home/borolebhushan8/bugbug/scripts/commit_classifier.py", line 596, in classify_regressor
>     commits[-1],
> IndexError: tuple index out of range
> ```

In the full output. this line also gets printed
`2020-08-03 16:25:04,928:INFO:bugbug.repository:No commits to analyze`

**Comment by marco-c** [comment]
` 2020-08-03 16:25:04,928:INFO:bugbug.repository:No commits to analyze`

Definitely seems like a problem; is it happening also on master (that is, without your patch)?

**Comment by bhushan-borole** [comment]
> ` 2020-08-03 16:25:04,928:INFO:bugbug.repository:No commits to analyze`
> 
> Definitely seems like a problem; is it happening also on master (that is, without your patch)?

Yes getting the same error

**Comment by marco-c** [comment]
Are both commands failing or just one of the two?

Try again the first command with a different diff ID (e.g. 322771), and the second command with a different revision (e.g. c8123a3e42497aaa892109bdbb414862a40c9747).

**Comment by bhushan-borole** [comment]
> c8123a3e42497aaa892109bdbb414862a40c9747

Getting this error after running the second command

```
Traceback (most recent call last):
  File "<string>", line 1, in <module>
  File "/home/borolebhushan8/bugbug/scripts/commit_classifier.py", line 786, in main
    classifier = CommitClassifier(
  File "/home/borolebhushan8/bugbug/scripts/commit_classifier.py", line 214, in __init__
    self.past_failures_data = test_scheduling.get_past_failures("label")
TypeError: get_past_failures() missing 1 required positional argument: 'readonly'
```

After changing the diff id:

```
Traceback (most recent call last):
  File "<string>", line 1, in <module>
  File "/home/borolebhushan8/bugbug/scripts/commit_classifier.py", line 786, in main
    classifier = CommitClassifier(
  File "/home/borolebhushan8/bugbug/scripts/commit_classifier.py", line 168, in __init__
    self.apply_phab(hg, phabricator_deployment, diff_id)
  File "/home/borolebhushan8/bugbug/scripts/commit_classifier.py", line 379, in apply_phab
    hg.import_(
  File "/home/borolebhushan8/.local/lib/python3.8/site-packages/hglib/client.py", line 996, in import_
    self.rawcommand(args, prompt=prompt, input=input)
  File "/home/borolebhushan8/.local/lib/python3.8/site-packages/hglib/client.py", line 263, in rawcommand
    raise error.CommandError(args, ret, out, err)
hglib.error.CommandError: (255, b'applying patch from stdin', b"unable to find 'mailnews/base/prefs/content/accountUtils.js' for patching\n(use '--prefix' to apply patch relative to the current directory)\n2 out of 2 hunks FAILED -- saving rejects to file mailnews/base/prefs/content/accountUtils.js.rej\nunable to find 'mail/extensions/openpgp/content/ui/enigmailKeyManager.xhtml' for patching\n(use '--prefix' to apply patch relative to the current directory)\n1 out of 1 hunks FAILED -- saving rejects to file mail/extensions/openpgp/content/ui/enigmailKeyManager.xhtml.rej\nunable to find 'mail/extensions/openpgp/content/ui/enigmailKeyManager.js' for patching\n(use '--prefix' to apply patch relative to the current directory)\n2 out of 2 hunks FAILED -- saving rejects to file mail/extensions/openpgp/content/ui/enigmailKeyManager.js.rej\nabort: patch failed to apply")
```

**Comment by marco-c** [comment]
Sorry, use this diff ID instead: 322778.

Regarding the first error, I'll fix it on master and then you can rerun it after rebasing on master.

**Comment by bhushan-borole** [comment]
> Sorry, use this diff ID instead: 322778.
> 
> Regarding the first error, I'll fix it on master and then you can rerun it after rebasing on master.

Getting the same error on both master and fork

**Comment by marco-c** [comment]
> Getting the same error on both master and fork

I hadn't pushed the fix yet, you can try again now (after rebasing on top of master).

Did you try the other command with the other diff ID?

**Comment by bhushan-borole** [comment]
> > Getting the same error on both master and fork
> 
> I hadn't pushed the fix yet, you can try again now (after rebasing on top of master).
> 
> Did you try the other command with the other diff ID?

Yes I tried the new diff ID, still getting the same error.

**Comment by bhushan-borole** [comment]
Getting the same error for both commands, on master as well as on fork

**Comment by marco-c** [comment]
I'm a bit confused, exactly which error are you now seeing?

**Comment by bhushan-borole** [comment]
> I'm a bit confused, exactly which error are you now seeing?

This one
```
Traceback (most recent call last):
  File "<string>", line 1, in <module>
  File "/home/borolebhushan8/bugbug/scripts/commit_classifier.py", line 795, in main
    classifier.classify(args.runnable_jobs)
  File "/home/borolebhushan8/bugbug/scripts/commit_classifier.py", line 587, in classify
    self.classify_regressor(commits)
  File "/home/borolebhushan8/bugbug/scripts/commit_classifier.py", line 596, in classify_regressor
    commits[-1],
IndexError: tuple index out of range
```

**Comment by marco-c** [comment]
I can't reproduce the error on master with this command:
```
BUGBUG_TEST_FAILURE_CONFIDENCE_THRESHOLD="0.7" BUGBUG_TEST_SELECTION_CONFIDENCE_THRESHOLD="0.3" BUGBUG_PHABRICATOR_URL='https://phabricator.services.mozilla.com/api/' BUGBUG_PHABRICATOR_TOKEN="api-mytokenhere" PYTHONPATH=. python3 -c "from scripts import commit_classifier; commit_classifier.main()" testlabelselect path/to//mozilla-central --revision='c8123a3e42497aaa892109bdbb414862a40c9747' --runnable-jobs='https://firefox-ci-tc.services.mozilla.com/api/queue/v1/task/YZOfqacTTIyUapQK3XrluQ/runs/0/artifacts/public/runnable-jobs.json'
```

**Comment by bhushan-borole** [comment]
> I can't reproduce the error on master with this command:
> 
> ```
> BUGBUG_TEST_FAILURE_CONFIDENCE_THRESHOLD="0.7" BUGBUG_TEST_SELECTION_CONFIDENCE_THRESHOLD="0.3" BUGBUG_PHABRICATOR_URL='https://phabricator.services.mozilla.com/api/' BUGBUG_PHABRICATOR_TOKEN="api-mytokenhere" PYTHONPATH=. python3 -c "from scripts import commit_classifier; commit_classifier.main()" testlabelselect path/to//mozilla-central --revision='c8123a3e42497aaa892109bdbb414862a40c9747' --runnable-jobs='https://firefox-ci-tc.services.mozilla.com/api/queue/v1/task/YZOfqacTTIyUapQK3XrluQ/runs/0/artifacts/public/runnable-jobs.json'
> ```

I cloned the master once again, and tried this command, but it is giving me the same error. I am also trying to figure out why.

**Comment by bhushan-borole** [comment]
Hey @marco-c 
Any ideas how to resolve that error?

**Comment by marco-c** [comment]
Unfortunately I can't reproduce it, so I can't easily tell what it could be.
I suggest adding some logging in https://github.com/mozilla/bugbug/blob/ab9c73a5eb76bf761399053fb95a9bc39f36ed58/bugbug/repository.py#L1006 to figure out why it's returning an empty list.

**Comment by marco-c** [comment]
Feel free to reopen if you still intend to work on this.

**Pull request closed without merging**